skiko/src/jvmMain/kotlin/org/jetbrains/skiko/cleanup.kt

```diff
@@ -0,0 +1,47 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import java.nio.file.Path
+import java.time.Duration
+import java.time.Instant
+import kotlin.concurrent.thread
+import kotlin.io.path.*
+
+/**
+ * Will asynchronously cleanup 'stale' entries in the [SkikoProperties.dataPath]:
+ * If directories are older than [SkikoProperties.dataCleanupDays], then they will be deleted.
+ */
+internal fun launchSkikoDataDirCleanupIfNecessary(dataDir: Path) {
+    thread(
+        name = ".skiko cleanup",
+        priority = Thread.MIN_PRIORITY
+    ) {
+        try {
+            withDirectoryLock(dataDir) {
```

Comment:
If we acquire the lock correctly (see my comment above), we can have a situation when the unpacking logic waits for the cleaning thread, which means it waits for a potentially long file operation.